client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx

```diff
@@ -0,0 +1,208 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React from 'react';
+import { localize } from 'i18n-calypso';
+import { isEmpty } from 'lodash';
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FoldableCard from 'components/foldable-card';
+import SectionHeader from 'components/section-header';
+import { checkInboundTransferStatus } from 'lib/domains';
+
+class TransferDomainPrecheck extends React.PureComponent {
+	static propTypes = {
+		domain: PropTypes.string,
+		setValid: PropTypes.func,
+	};
+
+	state = {
+		unlocked: false,
+		privacy: false,
+		email: '',
+		loading: true,
+	};
+
+	componentWillMount() {
+		this.refreshStatus();
+	}
+
+	componentWillUpdate( nextProps ) {
+		if ( nextProps.domain !== this.props.domain ) {
+			this.refreshStatus();
+		}
+	}
+
+	onClick = () => {
+		this.props.setValid( this.props.domain );
+	};
+
+	refreshStatus = () => {
+		this.setState( { loading: true } );
+
+		checkInboundTransferStatus( this.props.domain, ( error, result ) => {
+			if ( ! isEmpty( error ) ) {
+				return;
+			}
+
+			this.setState( {
+				email: result.admin_email,
+				privacy: false,
```

Comment:
You're not updating `privacy` based on `result.privacy` here :)

Also `privacy == true` means it does _not_ have privacy? Wut?